src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -89,6 +89,7 @@ def get(url, headers)
   def self.terminate
     unless self.client.nil?
       self.client.close
+      self.client = nil
```

Comment:
Hmm, now that I think about it, we made some changes around this stuff in the stable and master branches that hasn't gotten back to 2.1.x yet - https://github.com/puppetlabs/puppet-server/blob/master/src/ruby/puppet-server-lib/puppet/server/http_client.rb#L89-L93:

~~~ruby
def self.terminate
  unless @client.nil?
    @client.close
  end
end
~~~~

Before that, a `self.client.nil?` call would cause the client to be created if it hadn't already - and we'd turn right around and throw it away.  The change above was just an optimization for that case.

I'm okay with your `client=`, if that's what you'd prefer.  But I think it would be good to end up with the same implementations for these methods on all of our branches at the end of all of this.